components/txn_types/src/lock.rs

```diff
@@ -84,6 +84,10 @@ pub struct Lock {
     pub txn_size: u64,
     pub min_commit_ts: TimeStamp,
     pub use_async_commit: bool,
+    // This field is only valid for in-memory locks and does not need to be persisted because:
+    //   1. the lock should be converted to a write directly when 1pc succeeds.
+    //   2. the field should be reverted to false (default value) when 1pc fails.
```

Comment:
Perhaps we need to reset the flag to false in `fallback_1pc_locks` .

Reply:
Since the field is not persisted, it just like the flag automatically gets reverted when calling `lock.to_bytes` (fallback_1pc_locks -> txn.put_lock -> lock.to_bytes). Maybe just add a comment to `fallback_1pc_locks`?

Reply:
That approach is fine as well. We need a mechanism to inform code maintainers about this, minimizing deeply implicit dependencies or constraints.

Reply:
Personally I prefer setting it to false explicitly, in which way I think the semantics of the code can be more clear

Reply:
OK, I will update it later.